### PR TITLE
Configured Dovecot to log into its own logfile

### PR DESCRIPTION
**Pull request by dscoduc**
Added log path info into dovecot configuration file, added a config into logrotate.d folder, created a placeholder file in /var/log and set owner/permissions on the file.

### CODE REVIEW

File: setup/mail-dovecot.sh [hunk 1]
```diff
@@ -38,7 +38,26 @@ apt_install \
 # would be 20 users). Set it to 250 times the number of cores this
 # machine has, so on a two-core machine that's 500 processes/100 users).
 tools/editconf.py /etc/dovecot/conf.d/10-master.conf \
-	default_process_limit=$(echo "`nproc` * 250" | bc)
+	default_process_limit=$(echo "`nproc` * 250" | bc) \
+	log_path = /var/log/dovecot.log
+
+# Add logrotate entry for dovecot
+cat > /etc/dovecot/conf.d/90-plugin-fts.conf << EOF;
```
Comment (JoshData) [inline setup/mail-dovecot.sh]: Wrong path?

**Comment by dscoduc** [comment]
Indeed it was the wrong path for logrotate.d folder... corrected...

File: setup/mail-dovecot.sh [hunk 2]
```diff
@@ -38,7 +38,26 @@ apt_install \
 # would be 20 users). Set it to 250 times the number of cores this
 # machine has, so on a two-core machine that's 500 processes/100 users).
 tools/editconf.py /etc/dovecot/conf.d/10-master.conf \
-	default_process_limit=$(echo "`nproc` * 250" | bc)
+	default_process_limit=$(echo "`nproc` * 250" | bc) \
+	log_path = /var/log/dovecot.log
```
Comment (JoshData) [inline setup/mail-dovecot.sh]: Can't have spaces around the `=` here.

**Comment by JoshData** [comment]
Actually, why don't we log to mail.log instead?

**Comment by dscoduc** [comment]
That probably makes more sense…  I’ve corrected and checked in the changes…

**Comment by JoshData** [comment]
Thanks. The other thing to consider is that this probably breaks fail2ban monitoring of failed imap logins, since it's probably reading syslog for that.

**Comment by dscoduc** [comment]
According to /etc/fail2ban/jail.conf:


> [courierauth]
> 
> enabled  = false
> port     = smtp,ssmtp,submission,imap2,imap3,imaps,pop3,pop3s
> filter   = courierlogin
> logpath  = /var/log/mail.log

There is a port profile for adding imap and pop3 to the config...

**Comment by JoshData** [comment]
This looks fine, but it'll have to wait until I get a chance to test that it doesn't break fail2ban.

**Comment by JoshData** [comment]
Merged via fac8477. Thanks!

**Pull request closed without merging**